src/basic/architecture.c

```diff
@@ -18,7 +18,8 @@ Architecture uname_architecture(void) {
          * clean it up and break down the confusion on x86 and arm in particular.
          *
          * We try to distinguish CPUs, not CPU features, i.e. actual architectures that
-         * have genuinely different code. */
+         * have genuinely different code. This includes different userspace ABI, as used
+         * for LIB_ARCH_TUPLE and defined in https://wiki.debian.org/Multiarch/Tuples */
```

Comment:
sorry, but we aren't debian, we aren't just doign whatever nonsense debian wants.

if you want, then introduce another enum that identifies abis, but please, don't just funnel all that historic crap into a currently very usable architecture field.

Reply:
Usable for what, exactly? Again, your suggestion of faking a userspace ABI for another just cannot work. Try and build an x32 systext, mark it as x86_64 and attach it to your x86_64 system.

Reply:
or actually don't, we actually have an ABI identifier already: LIB_ARCH_TUPLE! It's 1:1 the debian thing. Why do we need to duplicate that in the Architecture enum? I don't get it?

I *explicitly* designed Architecture to not be about ABI, I added comments around it, to make this clear. I am of the very strong opinion that this should not be changed.

Reply:
This is needed in the preprocessor macro that generates the GPT macros, so unfortunately it cannot be the LIB_ARCH_TUPLE